File: cpp/src/patterns/state/state/SoldOutState.cc
#include "SoldOutState.h"
#include "VendingMachine.h"

SoldOutState::SoldOutState(VendingMachine *machine) {
    machine_ = machine;
}

void SoldOutState::Insert() {
    cout << "The machine is sold out" << endl;
}

void SoldOutState::Eject() {
    cout << "You haven't inserted money" << endl;
}

void SoldOutState::Bought() {
    cout << "You turned, but there are no goods" << endl;
}

void SoldOutState::Dispense() {
    cout << "No dispensed" << endl;
} 

File: Java/src/patterns/state/stateInterface/SoldOutState.java
package patterns.state.stateInterface;

public class SoldOutState implements State {
	private VendingMachine machine;
	
	public SoldOutState(VendingMachine machine) {
		this.machine = machine;
	}
	
	@Override
	public void insert() {
		System.out.println("The machine is sold out");
	}

	@Override
	public void eject() {
		System.out.println("You haven't inserted money");
	}

	@Override
	public void bought() {
		System.out.println("You turned, but there are no goods");
	}

	@Override
	public void dispense() {
		System.out.println("No dispensed");
	}
}
